### DISCUSSION TITLE
Deploying to GitHub Packages from GitHub Actions returns Unauthorized Error

### DISCUSSION THREAD

**Discussion by MRtecno98** [community/community#26499] [Actions]
<p>I have setup a workflow for building my maven project and deploying the artifact to Github Packages, but altrough I use the github token given by Actions, the deploy fails with error 401(Unauthorized). I tested it locally and it works flawlessly using my personal access token. The repo in question is MRtecno98/SpigotCommons.</p>
<p>The workflow code is the following:</p>
<pre><code>name: builds

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
       - uses: actions/checkout@v1

       - name: Set up JDK 1.8
         uses: actions/setup-java@v1
         with:
           java-version: 1.8

       - name: Build with Maven
         run: mvn -B clean package -DskipTests --file pom.xml

       - name: Run unit tests
         run: mvn -B test --file pom.xml

       - name: Deploy to GitHub packages
         run: mvn -B deploy --file pom.xml
         env:
            github-token: ${{ github.token }}
</code></pre>
<p>I’m also using a maven extension for using multiple settings.xml files, the second settings file contains this code:</p>
<pre><code>&lt;settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      http://maven.apache.org/xsd/settings-1.0.0.xsd"&gt;

  &lt;activeProfiles&gt;
    &lt;activeProfile&gt;github&lt;/activeProfile&gt;
  &lt;/activeProfiles&gt;

  &lt;profiles&gt;
    &lt;profile&gt;
      &lt;id&gt;github&lt;/id&gt;
      &lt;repositories&gt;
        &lt;repository&gt;
          &lt;id&gt;central&lt;/id&gt;
          &lt;url&gt;https://repo1.maven.org/maven2&lt;/url&gt;
          &lt;releases&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/releases&gt;
          &lt;snapshots&gt;&lt;enabled&gt;true&lt;/enabled&gt;&lt;/snapshots&gt;
        &lt;/repository&gt;

        &lt;repository&gt;
          &lt;id&gt;github&lt;/id&gt;
          &lt;name&gt;GitHub MRtecno98 Apache Maven Packages&lt;/name&gt;
          &lt;url&gt;https://maven.pkg.github.com/MRtecno98/SpigotCommons&lt;/url&gt;
        &lt;/repository&gt;
      &lt;/repositories&gt;
    &lt;/profile&gt;
  &lt;/profiles&gt;

  &lt;servers&gt;
  	&lt;server&gt;
		&lt;id&gt;github&lt;/id&gt;
	    &lt;username&gt;MRtecno98&lt;/username&gt;
	    &lt;password&gt;${env.github-token}&lt;/password&gt;
	&lt;/server&gt;
  &lt;/servers&gt;
&lt;/settings&gt;
</code></pre>
<p>As I already said, when I try to deploy locally(providing the access token with -D arguments) it deploys normally to Packages, it seems to fail just when Actions executes it.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/mrtecno98">@mrtecno98</a> ,</p>
<p>I strongly recommend you use ’ <strong>_</strong>’ character instead of ’ <strong>-</strong>’ character when you name the environment variables. As I known, in almost all cases, the ’ <strong>-</strong>’ character is not supported to name the environment variables.<br>
Maybe the below example can explain this point more intuitively:</p>
<pre><code>- name: Test env
  run: |
    echo "test_env01 = $test_env01"
    echo "test-env02 = $test-env02"
  env:
    test_env01: 'Test env 01'
    test-env02: 'Test env 02'
</code></pre>
<p>the output:</p>
<pre><code>test_env01 = Test env 01
test-env02 = -env02
</code></pre>
<p>You can see the output of the second environment variable " <strong>test-env02</strong>" is incorrect, it only recognizes the string before the ’ <strong>-</strong>’ character as the name of the environment variable.</p>
<p>So, if you want to correctly set the github token as an environment variable, you can use one of the following ways:</p>
<pre><code>- name: Publish to GitHub Packages Apache Maven
      run: |
        mvn deploy -s settings.xml
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>or</p>
<pre><code>- name: Publish to GitHub Packages Apache Maven
      run: |
        echo $PACKAGE_VERSION
        mvn deploy -s settings.xml
      env:
        GITHUB_TOKEN: ${{ github.token }}
</code></pre>
<p>I generally use uppercase letters to name environment variables. Of course, you also can keep using the lowercase letters if you want.</p>

**Comment by MRtecno98** [comment]
<p>That was it! Thanks a lot!</p>
